HeterogeneousCore/SonicTriton/src/TritonService.cc

```diff
@@ -238,7 +247,67 @@ void TritonService::preBeginJob(edm::PathsAndConsumesOfModulesBase const&, edm::
   if (fallbackOpts_.wait >= 0)
     fallbackOpts_.command += " -w " + std::to_string(fallbackOpts_.wait);
   for (const auto& [modelName, model] : unservedModels_) {
-    fallbackOpts_.command += " -m " + model.path;
+    //We need to edit the triton config files, need to copy from cvmfs
+    std::string delimiter = "/";
+    std::string tmp_model_path = model.path;
+    size_t pos = 0;
+    std::string token;
+    std::string model_dir;
+    while ((pos = tmp_model_path.find(delimiter)) != std::string::npos) {
+      if (tmp_model_path.substr(0, pos) != "config.pbtxt") {
+        token = tmp_model_path.substr(0, pos);
+        model_dir += token + "/";
+      }
+      tmp_model_path.erase(0, pos + delimiter.length());
+    }
+
+    std::string remover = "mkdir local_model_repo; rm -rf local_model_repo/" + token;
+    std::string copier = "cp -r ";
+    const auto& [output, rv] = execSys(remover + "; " + copier + model_dir + " local_model_repo/");
+    fallbackOpts_.command += " -m ./local_model_repo/" + token + "/config.pbtxt";
+
+    //Need to edit files differently depending on model framework
+    bool isONNX = false;
+    bool isTF = false;
+
+    size_t offset;
+    std::string line;
```

Comment:
Please move this declaration inside the `if` block as it is needed only there.

Reply:
Yes, all deleted and moved to cmsTriton script